core/src/main/java/smashdudes/gameplay/state/playerstate/ground/GroundRunningState.java

```diff
@@ -18,14 +23,18 @@ public void innerUpdate(float dt)
         CharacterInputComponent ci = entity.getComponent(CharacterInputComponent.class);
         VelocityComponent v  = entity.getComponent(VelocityComponent.class);
 
-        float speed = v.runSpeed;
+        float speed = 0;
```

Comment:
No effective change?
Might lead to more branch misprediction?

Reply:
Branch misprediction is not a concern at all.

There is a logical change. We only update the velocity if it is non-zero. 
The old logic would set the velocity to zero if we were not pressing a key, so the deceleration did nothing.